rs/ledger_suite/icp/tests/golden_nns_state.rs

```diff
@@ -38,11 +46,15 @@ fn should_create_state_machine_with_golden_nns_state() {
     // Verify ledger, archives, and index block parity
     setup.verify_ledger_archive_index_block_parity();
 
+    setup.perform_transactions();
+
     // Downgrade all the canisters to the mainnet version
     setup.downgrade_to_mainnet();
 
     // Verify ledger, archives, and index block parity
     setup.verify_ledger_archive_index_block_parity();
+
+    setup.perform_transactions();
```

Comment:
It might be worth to add an additional `setup.verify_ledger_archive_index_block_parity();` after final tx generation.